src/core/trtserver.h

```diff
@@ -169,6 +169,21 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_SharedMemoryBlockGpuNew(
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_SharedMemoryBlockDelete(
     TRTSERVER_SharedMemoryBlock* shared_memory_block);
 
+/// Get the memory type of a shared memory block object.
+/// \param shared_memory_block The object whose memory type is required.
+/// \param memory_type Returns the memory type of the shared memory block.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_Error* TRTSERVER_SharedMemoryBlockMemoryType(
+    TRTSERVER_SharedMemoryBlock* shared_memory_block,
+    TRTSERVER_Memory_Type* memory_type);
+
+/// Get the memory type id of a shared memory block object.
+/// \param shared_memory_block The object whose memory type is required.
+/// \param memory_type_id The CPU/GPU number the shared memory region is in.
```

Comment:
CPU number seems strange here. Currently we don't have meaning for `memory_type_id` in the case of system memory. Perhaps just say the device ID if the region is in GPU memory.

Reply:
Sure